library/core/src/main/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelector.java

```diff
@@ -2730,6 +2742,9 @@ public int compareTo(TextTrackScore other) {
       if (this.preferredLanguageScore != other.preferredLanguageScore) {
         return compareInts(this.preferredLanguageScore, other.preferredLanguageScore);
       }
+      if (this.preferredRoleFlagsScore != other.preferredRoleFlagsScore) {
```

Comment:
If everything else is exactly the same, and no role flag preference is provided in the parameters, we should probably still compare by the existing role flags and prefer the tracks *without* CAPTION or DESCRIBES_MUSIC_AND_SOUND role flags. This under the assumption that no specification means that non-captioning text tracks are preferred.  This comparison should be the very last one after the selectedAudioLanguageScore.